### PR TITLE
[WIP] Disable logging at runtime

### PR DESCRIPTION
**Pull request by damianh**
Addresses #50 

Exercised via `YourLib.Logging.LogProvider.IsLoggingEnabled = false`. The `LoggerExecutionWrapper` will check this each call. This means it's safe for both statically initialized logger fields and instance fields/ local instances.

@leastprivilege the issue problem with the Env Var approach is that it will disable Logging for _all_ libraries that use LibLog. Is this desirable / wise? If I could get {YourNamespace}_LIBLOG_DISABLE working, would that be better?

### CODE REVIEW

File: src/LibLog/LibLog.cs [hunk 1]
```diff
@@ -573,11 +590,14 @@ public bool Log(LogLevel logLevel, Func<string> messageFunc, Exception exception
     internal class LoggerExecutionWrapper : ILog
     {
         private readonly Logger _logger;
+        private readonly Func<bool> _getIsDisabled;
         internal const string FailedToGenerateLogMessage = "Failed to generate log message";
+        private const string EnvironmentVar = "$rootnamespace$_LIBLOG_DISABLE";
```
Comment (damianh) [inline src/LibLog/LibLog.cs]: Hoping this will be transformed by VS DTE. Will need to build a package to test it.

**Comment by damianh** [comment]
@leastprivilege This is on my CI feed https://www.myget.org/F/dh/  4.1.0-build00116 . Can you take it for a spin?

You're env var will look like this: https://github.com/damianh/LibLog/blob/master/src/LibLog.Example.Library/App_Packages/LibLog.4.0/LibLog.cs#L377

**Comment by leastprivilege** [comment]
My Powershell sucks - but this

`$env:LibLog.Example.Library_LIBLOG_DISABLE = "true"`

does not work.

How would I set the env variable?

**Comment by damianh** [comment]
try $env:{YourNamespace}_LIBLOG_DISABLE

**Comment by leastprivilege** [comment]
What's my namespace - base namespace?

$env:IdentityServer3_LIBLOG_DISABLE

or something else?

**Comment by leastprivilege** [comment]
why don't you add a global switch?

LIBLOG_DISABLE

**Comment by damianh** [comment]
Because that's an awfully big hammer and would disable across all libs, including 3rd party, that may not be desirable.

Where is the branch where you are trying this out?

**Comment by damianh** [comment]
You can find the name of the env var in your lib by looking for this https://github.com/damianh/LibLog/blob/master/src/LibLog.Example.Library/App_Packages/LibLog.4.1/LibLog.cs#L415

**Comment by leastprivilege** [comment]
`$env:{IdentityServer3.Core_LIBLOG_DISABLE} = "true"`

does not work either.

I like the per namespace disable - still i think there should be a global flag.

**Comment by leastprivilege** [comment]
At line:1 char:1
+ $env:{IdentityServer3.Core_LIBLOG_DISABLE} = "true"
+ ~~~~~
Variable reference is not valid. ':' was not followed by a valid variable name character. Consider using ${} to delimit the name.
At line:1 char:6
+ $env:{IdentityServer3.Core_LIBLOG_DISABLE} = "true"
+      ~
Unexpected token '{' in expression or statement.
At line:1 char:6
+ $env:{IdentityServer3.Core_LIBLOG_DISABLE} = "true"
+      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The assignment expression is not valid. The input to an assignment operator must be an object that is able to accept assignments, such as a
variable or a property.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvalidVariableReferenceWithDrive

**Comment by damianh** [comment]
Push this to a branch somewhere pls, so I can run it locally.

**Comment by leastprivilege** [comment]
I am just trying to set the env var in PS. but sure can do.

**Comment by leastprivilege** [comment]
https://github.com/IdentityServer/IdentityServer3/tree/ShutUpLibLog

**Comment by damianh** [comment]
lol

**Comment by leastprivilege** [comment]
yea - i think there should be a way to disable all logging - I really don't want it in my CI builds.

**Comment by damianh** [comment]
`${env:IdentityServer3.Core_LIBLOG_DISABLE"} = "true"`

**Comment by damianh** [comment]
image

**Comment by leastprivilege** [comment]
did that - still see log output

**Comment by leastprivilege** [comment]
run build.ps1

**Pull request merged**